phylogenetic/rules/merge_inputs.smk

```diff
@@ -0,0 +1,36 @@
+"""
+This part of the workflow merges inputs based on what is defined in the config.
+
+The config dict is expected to have a top-level `inputs` list that defines the
+separate inputs' name, metadata, and sequences. Optionally, the config can have
+a top-level `additional-inputs` list that is used to define additional data that
+are combined with the default inputs:
+
+```yaml
+inputs:
+    - name: default
+      metadata: <path-or-url>
+      sequences: <path-or-url>
+
+additional_inputs:
+    - name: private
+      metadata: <path-or-url>
+      sequences: <path-or-url>
+```
+
+The merged inputs can then be accessed in any downstream rules with the functions
+`input_metadata` and `input_sequences`.
+
+Example:
+
+    rule filter:
+        input:
+            metadata = input_metadata,
+            sequences = input_sequences,
+
+See zika for a concrete example of how the inputs are parsed and defined
+<https://github.com/nextstrain/zika/blob/main/phylogenetic/rules/merge_inputs.smk>
+
+See avian-flu for an example of using wildcards in inputs
+<https://github.com/nextstrain/avian-flu/blob/f1657094ebf9da39660a7f4ba6a5ddf51b6a9c3e/rules/main.smk#L93-L116>
```

Comment:
I'd not suggest referring to avian-flu without some serious caveats! That uses a lot of bespoke code to download remote files [1] and has a lot of complexity to handle per-segment FASTAs. 

[1] I should use my new `path_or_url` code, but this will make it snakemake v7 incompatible which I'm not sure is a good idea at the moment.